test/test_discovery_and_monitoring.py

```diff
@@ -408,17 +407,21 @@ def handle(self):
         self.request.close()
 
 
-class TestHeartbeatStartOrdering(unittest.TestCase):
-    def start_server(self, events):
-        server = socketserver.TCPServer(("localhost", 9999), MockTCPHandler)
-        server.events = events
-        server.handle_request()
-        server.server_close()
+class TCPServer(socketserver.TCPServer):
+    allow_reuse_address = True
 
+    def handle_request_and_shutdown(self):
+        self.handle_request()
+        self.server_close()
+
+
+class TestHeartbeatStartOrdering(unittest.TestCase):
     def test_heartbeat_start_ordering(self):
         events = []
         listener = HeartbeatEventsListListener(events)
-        server_thread = threading.Thread(target=self.start_server, args=(events,))
+        server = TCPServer(("localhost", 9999), MockTCPHandler)
```

Comment:
We need to create the TCPServer first to ensure the server is listening before creating the MongoClient.